### ISSUE TITLE
Prediction inaccurate despite low validation loss

### ISSUE THREAD

**Issue by nkma1989** [experiencor/basic-yolo-keras#154]
Hello.
I've been following the notebook implementation found here:
https://github.com/experiencor/basic-yolo-keras/blob/master/Yolo%20Step-by-Step.ipynb
I've been training on the raccoon dataset. After some fiddling I finally got the network to converge.
Using early stop i get the best epoch as follows.
```
Epoch 70/100
 9/10 [==========================>...] - ETA: 15s - loss: 0.00 - ETA: 13s - loss: 0.00 - ETA: 11s - loss: 0.00 - ETA: 9s - loss: 0.0038 - ETA: 8s - loss: 0.003 - ETA: 6s - loss: 0.003 - ETA: 4s - loss: 0.003 - ETA: 3s - loss: 0.003 - ETA: 1s - loss: 0.0038Epoch 00070: val_loss improved from 0.00387 to 0.00372, saving model to weights_raccoon.h5
10/10 [==============================] - 24s 2s/step - loss: 0.0038 - val_loss: 0.0037
```

I use the following to predict:
```
model.load_weights(r'path\weights_raccoon.h5')
image = cv2.imread(all_imgs[train_valid_split:][10]['filename'])
plt.figure(figsize=(10,10))
input_image = cv2.resize(image, (416, 416))
input_image = input_image / 255.
input_image = input_image[:,:,::-1]
input_image = np.expand_dims(input_image, 0)
netout = model.predict([input_image, dummy_array])
boxes = decode_netout(netout[0], 
                      obj_threshold=0.02,
                      nms_threshold=0.003,
                      anchors=ANCHORS, 
                      nb_class=CLASS)
image = draw_boxes(image, boxes, labels=LABELS)
plt.imshow(image[:,:,::-1]); plt.show()
```
All good, but the result is what confuses me. I had to set the thresholds low to even get bounding boxes. This is what I see in general:
![capture](https://user-images.githubusercontent.com/24934325/36385859-2593bb8e-1594-11e8-8d32-6c063835733a.JPG)
I cannot see where I went wrong. Tried many different parameter combinations and when i hit this low validation loss i thought I had it. What am I missing? Hopefully you can point me in the right direction.

Best Regards 
Niels

**Comment by LucasVandroux** [comment]
@nkma1989, it seems that we have a similar problem: really low loss but poor predictions (#157) I didn't find a solution yet but I am looking into it at the moment. Did you make any progress on your side?

**Comment by nkma1989** [comment]
@LucasVandroux Yes I did. What I learned the problem was, was that I was performing the warm up epochs. So what I did was follow the example using command prompt and it worked. You initially run the warm up epochs(I've been using 3). Save the weights. Set warm up epoch to 0. Load the weights you've just trained and then train the network again. This should also work in the notebook. However i have not tested it.
Hope it helps
